include/laser_filters/scan_shadows_filter.h

```diff
@@ -111,13 +124,33 @@ class ScanShadowsFilter : public filters::FilterBase<sensor_msgs::LaserScan>
       ROS_ERROR("max_angle must be max_angle <= 180. Forcing max_angle = 180.\n");
       max_angle_ = 180.0;
     }
+
     shadow_detector_.configure(
         angles::from_degrees(min_angle_),
         angles::from_degrees(max_angle_));
 
+    param_config.min_angle = min_angle_;
+    param_config.max_angle = max_angle_;
+    param_config.window = window_;
+    param_config.neighbors = neighbors_;
+    param_config.remove_shadow_start_point = remove_shadow_start_point_;
+    dyn_server_->updateConfig(param_config);
+
     return true;
   }
 
+  void reconfigureCB(ScanShadowsFilterConfig& config, uint32_t level)
+  {
```

Comment:
The reconfigure callback can be called from a different thread than the main processing code. Could you add a lock so that the parameters are never used in a half-updated state?